frontend/src/metabase/visualizations/click-actions/actions/HideColumnAction/HideColumnAction.tsx

```diff
@@ -37,24 +34,22 @@ export const HideColumnAction: LegacyDrill = ({
       tooltip: t`Hide column`,
       default: true,
       action: () => {
-        const columnSettings = getColumnSettingsWithRefs(
-          settings?.["table.columns"] || [],
-        );
-        const query = question.query();
-
-        const columnSettingIndex = findColumnSettingIndex(
-          query,
-          column,
+        const columnSettings = settings?.["table.columns"] ?? [];
+        const [columnSettingIndex] = findColumnSettingIndexesForColumns(
```

Comment:
This is still wrong and will match the wrong column setting in complex cases. To figure out in other PRs. We need to pass the whole column list to match properly.